src/Controls/src/Core/AppThemeBinding.cs

```diff
@@ -8,10 +8,38 @@ namespace Microsoft.Maui.Controls
 {
 	class AppThemeBinding : BindingBase
 	{
+		class AppThemeProxy : Element
+		{
+			public AppThemeProxy(Element parent, AppThemeBinding binding)
+			{
+				_parent = parent;
+				Binding = binding;
+				this.SetDynamicResource(AppThemeProperty, "__ApplicationTheme__");
+				((IElementDefinition)parent).AddResourcesChangedListener(OnParentResourcesChanged);
+			}
+
+			public static BindableProperty AppThemeProperty = BindableProperty.Create("AppTheme", typeof(AppTheme), typeof(AppThemeBinding), AppTheme.Unspecified,
+				propertyChanged: (bindable, oldValue, newValue) => ((AppThemeProxy)bindable).OnAppThemeChanged());
+			readonly Element _parent;
+
+			public void OnAppThemeChanged()
+			{
+				Binding.Apply(false);
```

Comment:
I'm curious about the performance impact, especially on CollectionView scrolling. Could we perhaps have an UITest with this example https://github.com/dotnet/maui/issues/18505 where we can display the minimum fps in a Label by scrolling and verifying that the value is greater than n (where n is value higher than before)?